### PR TITLE
#452 Add UI responsiveness to the main title, the ProgramPicker and the Calendar in the Program page

### LINKED ISSUE #452: Add UI responsiveness to the main title, the program picker, the calendar and pop-up in the Program page
## Expected behaviour

The main title and the ProgramPicker should be placed on separate horizontal lines on a small screen size.
The main title should be right-aligned on a small screen size.
The calendar should have day view on a small screen size.
The time in the ActivityDialog should be placed on separate line on a small screen size.

## Actual behaviour

The main title and the ProgramPicker is placed in one line. It looks good on tablet and desktop screens, but needs re-flow on a small screen.
The calendar looks the same on all screen sizes.
The time of activity in the ProgramActivityDialog needs re-flow on small screen size.

## Details and resources

Mobile breakpoint (min-width: 360px)
* The title of the page should be on a separate horizontal line.
* The Program Picker should be on a separate horizontal line after title of the page
* The width of the Program Picker's input should be less than screen size.
* The font-size of the Program Pickers should be smaller than for the desktop screen.
* The calendar should display only day view of the calendar. 
* Buttons of the month/week/agenda shouldn't be displayed.
* The time of the activity in ProgramActivityDialog should be displayed on a separate horizontal line. 

Screenshot 2022-11-28 at 15 17 10

Screenshot 2022-11-28 at 15 25 29

Tablet breakpoint (min-width: 768px) and Desktop breakpoint (min-width: 1440px)
*UI looks good. 

## Checklist

- [x] This issue is linked to the appropriate project.
- [x] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by veranika-karpava**
Authored by: @veranika-karpava

## Proposed changes

This pull request resolves #452 by:
* Displaying the main title, ProgramPicker and Calendar in rows on small screens and in columns on medium and large screens.
* Adding responsiveness to the ProgramPicker using different font-sizes and widths for different screens.
* Changing font-size for different screens in the Toolbar
* Displaying only day view calendar on small screens
* Not displaying buttons of view option on small screens
* Changing font-size of the TimeRange in ActivityDialog for different screens

Small screen (min-width: 375px and max-width: 600px)
min-width: 375px :
Screenshot 2022-12-06 at 13 46 29
Screenshot 2022-12-06 at 13 53 44

max-width: 600px:
Screenshot 2022-12-06 at 13 49 17
Screenshot 2022-12-06 at 13 54 33

greater than 600px:
Screenshot 2022-12-06 at 13 50 07
Screenshot 2022-12-06 at 13 55 14

## Checklist

- [x] Are the issues being addressed linked to this PR?
- [x] Do all commit messages start with the issue number?
- [x] Are all code changes sufficiently tested?
- [x] Are there screenshots for UI changes?

### CODE REVIEW

**Review by wendyyng — APPROVED** [review]
I git checkout to this branch and the calendar looks great on different dimensions/devices. Running yarn delint also didn't change any formatting.

I approve this PR and great job!

File: webapp/src/components/ProgramCalendar.tsx [hunk 1]
```diff
@@ -45,7 +45,18 @@ const ProgramCalendar = ({ program }: ProgramCalendarProps) => {
     end: new Date(),
     programTitle: '',
   });
-  const [currentView, setCurrentView] = React.useState<string>(Views.WEEK);
+  const [currentWidth, setCurrentWidth] = React.useState<number>(
+    window.innerWidth
+  );
+  const [currentView, setCurrentView] = React.useState<any>(
+    currentWidth <= 600 ? Views.DAY : Views.WEEK
```
Comment (camianderson) [inline webapp/src/components/ProgramCalendar.tsx:52]: When I was trying to test the functionality in my local machine it gives me an error and display an empty page if I am in the week view and make my window smaller than 600px. If I refresh it shows the daily view again...
have you noticed that?

https://user-images.githubusercontent.com/98124157/206277543-615988d1-0794-46d7-bab4-da6c71eb4d3d.mov

Comment (veranika-karpava) [reply webapp/src/components/ProgramCalendar.tsx:52]: I noticed it when I resized the screen very fast. I'm not sure why it doesn't work. Anyway, I'm trying to refactor the code. Thank you for letting me know.

**Comment by SanazDn** [comment]
Great job! Everything works as expected in different screens. But the only thing I am getting error in the console for "Please update the following components: DayColumn".

**Comment by phoebehala** [comment]
Nice work! The responsive worked as expected on my end.
Before merging into main, please bring this branch update to date with main :)

**Review by phoebehala — APPROVED** [review]

**Comment by veranika-karpava** [comment]
@SanazDn Thank you for asking about this warning. I saw this warning as well. This something that we couldn't change because pre-build package. I asked about it Spencer.

File: webapp/src/components/ProgramCalendar.tsx (hunk 1 above)
Comment (rrabinovitch) [reply webapp/src/components/ProgramCalendar.tsx]: This is happening on my end as well

Comment (veranika-karpava) [reply webapp/src/components/ProgramCalendar.tsx]: @rrabinovitch It's weird. I updated  code and it should be work now. Could you describe when it was happen?

**Pull request merged**